src/jit/compiler.cpp

```diff
@@ -540,7 +540,7 @@ var_types    Compiler::argOrReturnTypeForStruct(unsigned size, CORINFO_CLASS_HAN
         break;
 
     default:
-#if FEATURE_MULTIREG_RET
+#if FEATURE_MULTIREG_RET && !defined(_TARGET_X86_)
```

Comment:
MAX_RET_MULTIREG_BYTES should be set to 2 for x86, since it uses 2 registers to return longs (right?) The problem is that this method is assuming that a multireg return is a struct. That should either be fixed, or the code should be factored into a 32-bit version and a 64-bit version (which I think would be cleaner).